src/code/z_camera.c

```diff
@@ -574,71 +574,86 @@ s16 Camera_XZAngle(Vec3f* to, Vec3f* from) {
     return CAM_DEG_TO_BINANG(RAD_TO_DEG(Math_FAtan2F(from->x - to->x, from->z - to->z)));
 }
 
-s16 func_80044ADC(Camera* camera, s16 yaw, s16 arg2) {
-    static f32 D_8015CE50;
-    static f32 D_8015CE54;
-    static CamColChk D_8015CE58;
+s16 Camera_GetPitchAdjFromFloorHeightDiffs_(Camera* camera, s16 viewYaw, s16 initAndReturnZero) {
+    static f32 sFloorYSmallOffsetForwards;
+    static f32 sFloorYBigOffsetForwards;
+    static CamColChk sCamColChk_;
     Vec3f playerPos;
-    Vec3f rotatedPos;
+    Vec3f playerPosSmallOffsetForwards;
     Vec3f floorNorm;
-    f32 temp_f2;
-    s16 temp_s0;
-    s16 temp_s1;
-    f32 phi_f18;
-    f32 sinYaw;
-    f32 cosYaw;
+    f32 playerPosOffsetY;
+    s16 pitchSmallOffset;
+    s16 pitchBigOffset;
+    f32 floorYDiffBigOffsetForwards;
+    f32 viewForwardsUnitX;
+    f32 viewForwardsUnitZ;
     s32 bgId;
-    f32 sp30;
-    f32 sp2C;
-    f32 phi_f16;
+    f32 smallOffsetXZ;
+    f32 bigOffsetXZ;
+    f32 floorYDiffSmallOffsetForwards;
     f32 playerHeight;
 
-    sinYaw = Math_SinS(yaw);
-    cosYaw = Math_CosS(yaw);
+    viewForwardsUnitX = Math_SinS(viewYaw);
+    viewForwardsUnitZ = Math_CosS(viewYaw);
+
     playerHeight = Player_GetHeight(camera->player);
-    temp_f2 = CAM_DATA_SCALED(OREG(19)) * playerHeight;
-    sp30 = CAM_DATA_SCALED(OREG(17)) * playerHeight;
-    sp2C = CAM_DATA_SCALED(OREG(18)) * playerHeight;
+    playerPosOffsetY = CAM_DATA_SCALED(OREG(19)) * playerHeight; // 1.2 *
+    smallOffsetXZ = CAM_DATA_SCALED(OREG(17)) * playerHeight;    // 1 *
+    bigOffsetXZ = CAM_DATA_SCALED(OREG(18)) * playerHeight;      // 2.5 *
```

Comment:
I'd like to start naming these `OREG()` variables eventually, but that can be done separately if you'd like